### PR TITLE
Refactor ZerionAPI module and migrate Positions and Portfolio

### PR DESCRIPTION
**Pull request by everdimension**

### CODE REVIEW

File: src/modules/zerion-api/requests/ResponseBody.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export type ResponseBody<T> = { data: T | null; errors?: { title: string } };
```
Comment (zerts) [inline src/modules/zerion-api/requests/ResponseBody.ts:1]: `export type ResponseBody<T> = { data: T | null; errors?: { title: string, detail: string } };`

File: src/ui/shared/user-address/useAddressParams.ts [hunk 2]
```diff
@@ -1,11 +1,10 @@
-import type { AddressParams } from 'defi-sdk';
 import { useMemo } from 'react';
 import { useQuery } from '@tanstack/react-query';
 import { normalizeAddress } from 'src/shared/normalizeAddress';
 import { walletPort } from 'src/ui/shared/channels';
 
 interface Result {
-  params: AddressParams;
+  params: { address: string };
```
Comment (zerts) [inline src/ui/shared/user-address/useAddressParams.ts:7]: what about multiple addresses case?

File: src/ui/pages/Overview/Positions/Positions.tsx [hunk 3]
```diff
@@ -816,28 +800,22 @@ export function Positions({
...
-    value: portfolioDecomposition,
-    isLoading: portfolioDecompositionIsLoading,
-  } = useAddressPortfolioDecomposition(
-    {
-      address: singleAddressNormalized,
-      currency,
-    },
-    { enabled: ready, client: useDefiSdkClient() }
+  const { data, ...portfolioQuery } = useWalletPortfolio(
+    { addresses: [params.address], currency },
```
Comment (zerts) [inline src/ui/pages/Overview/Positions/Positions.tsx:804]: Is it ok that we changed `singleAddressNormalized` to `params.address`?

File: src/ui/pages/SwapForm/SwapForm.tsx [hunk 4]
```diff
@@ -1,10 +1,5 @@
 import { useSelectorStore } from '@store-unit/react';
-import {
-  client,
-  useAddressMembership,
-  useAddressPortfolioDecomposition,
-  useAddressPositions,
-} from 'defi-sdk';
+import { client, useAddressMembership, useAddressPositions } from 'defi-sdk';
```
Comment (zerts) [inline src/ui/pages/SwapForm/SwapForm.tsx:2]: Do not forget to update `useAddressPositions` here

**Comment by zerts** [comment]
Also don't forget to update `useAddressPositions` here: `src/ui/components/AllowanceView/AllowanceView.tsx`

File: src/modules/zerion-api/hooks/useWalletPositions.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    queryFn: async () => {
+      const response = await ZerionAPI.walletGetPositions(params, {
+        source,
+      });
+      return toAddressPositions(response);
+    },
+    suspense,
+    enabled,
+    keepPreviousData,
+    staleTime: 20000,
```
Comment (zerts) [inline src/modules/zerion-api/hooks/useWalletPositions.ts:32]: Do we want to add refetchInterval here as well? So when extension is opened, overview prices can be updated without extra actions?

**Review by zerts — APPROVED** [review]

File: src/modules/zerion-api/requests/get-gas-prices.ts [hunk 6]
```diff
@@ -1,11 +1,19 @@
...
 
-export interface Response {
+interface Response {
   data: ChainGasPrice;
   errors?: { title: string; detail: string }[];
 }
+
+export function getGasPrices({ chain }: Params, options?: ClientOptions) {
+  const params = new URLSearchParams({ chain: chain.toString() });
+  const endpoint = `chain/get-gas-prices/v1${params}`;
```
Comment (zerts) [inline src/modules/zerion-api/requests/get-gas-prices.ts:17]: const endpoint = `chain/get-gas-prices/v1?${params}`;

File: src/ui/pages/Overview/Positions/Positions.tsx (hunk 3 above)
Comment (everdimension) [reply src/ui/pages/Overview/Positions/Positions.tsx:804]: Yeah, they have essentially been the same

**Pull request merged**